### ISSUE TITLE
Incorrect Python version shown

### ISSUE THREAD

**Issue by wimglenn** [devpi/devpi#949]
[`devpi_common/metadata.py:get_pyversion_filetype`](
https://github.com/devpi/devpi/blob/16dc3cc4b98389b9504b6705e4bbb8975b273c1f/common/devpi_common/metadata.py#L63-L77) does not parse Python versions 3.10+ correctly by regex, for example:

```
>>> from devpi_common.metadata import get_pyversion_filetype
>>> get_pyversion_filetype("h5py-3.6.0-cp310-cp310-win_amd64.whl")
('3.1.0', 'bdist_wheel')
```

Similarly a cp311 wheel shows as **Python-3.1.1** in the UI.

It may be persisting some incorrect values at upload too:  https://github.com/devpi/devpi/blob/16dc3cc4b98389b9504b6705e4bbb8975b273c1f/client/devpi/upload.py#L182-L184

**Comment by fschulze** [comment]
It seems like parsing these isn't actually specified. For PyPy 2 it looks like "pp253" actually meant "PyPy 2.5.3", but for PyPy 3 they didn't use minor versions. I switch behaviour on whether the version starts with ``3`` for ``cp`` and ``pp`` tags.

**Comment by fschulze** [comment]
Should be fixed with devpi-common 3.7.2.
